apps/backend/src/auth/ProposalAuthorization.ts

```diff
@@ -224,19 +192,61 @@ export class ProposalAuthorization {
         ).some((value) => value)
       : false;
 
+    return isInternalReviewerOnSomeTechnicalReview;
+  }
+
+  async hasReadRights(
+    agent: UserWithRole | null,
+    proposalOrProposalId: Proposal | number
+  ): Promise<boolean> {
+    if (!agent) {
+      return false;
+    }
+
+    const proposal = await this.resolveProposal(proposalOrProposalId);
+
+    if (!proposal) {
+      return false;
+    }
+
+    const currentRole = agent?.currentRole?.shortCode;
+
+    let haveAccess = false;
+
+    switch (currentRole) {
+      case Roles.USER:
+        haveAccess =
+          (await this.isMemberOfProposal(agent, proposal)) ||
+          (await this.isVisitorOfProposal(agent, proposal.primaryKey));
+        break;
+      case Roles.INSTRUMENT_SCIENTIST:
+        haveAccess =
+          (await this.isInstrumentManagerToProposal(
+            agent,
+            proposal.primaryKey
+          )) || (await this.isScientistToProposal(agent, proposal.primaryKey));
+        break;
+      case Roles.INTERNAL_REVIEWER:
+        haveAccess = await this.isInternalReviewer(agent, proposal.primaryKey);
+        break;
+      case Roles.FAP_REVIEWER:
+      case Roles.FAP_SECRETARY:
+      case Roles.FAP_CHAIR:
+        haveAccess = await this.isMemberOfFapProposal(
+          agent,
+          proposal.primaryKey
+        );
+        break;
+      default:
+        break;
+    }
+
     return (
       this.userAuth.isUserOfficer(agent) ||
```

Comment:
Can't we have every check in the switch statement?(USER_OFFICE, SAMPLE_SAFETY_REVIEWER,...)